### DISCUSSION TITLE
Running Node.js in Background in Linux Throws Exception

### DISCUSSION THREAD

**Discussion by Power-tile** [requarks/wiki#3114] [Help / Questions] [answered]
I am running Wiki.js on a Ubuntu18.04 LTS system and had finished setting up the server. I can successfully access the wiki.js page after starting it with ```nodejs server``` in my SSH shell, so now I am trying out the background method suggested.

I pasted the following into ```/etc/systemd/system/wiki.service```, as suggested:
```
[Unit]
Description=Wiki.js
After=network.target

[Service]
Type=simple
ExecStart=/usr/bin/nodejs server
Restart=always
# Consider creating a dedicated user for Wiki.js here:
User=nobody
Environment=NODE_ENV=production
WorkingDirectory=/mnt/wiki

[Install]
WantedBy=multi-user.target
```
... and executed ```systemctl daemon-reload```, ```systemctl start wiki```, and ```systemctl enable wiki```.

However, the following exception is raised in my journalctl log:
```
Started Wiki.js.
Loading configuration from /mnt/wiki/config.yml... OK
internal/fs/utils.js:307
    throw err;
     ^
 Error: EACCES: permission denied, open '/mnt/wiki/package.json'
     at Object.openSync (fs.js:476:3)
     at Object.readFileSync (fs.js:377:35)
     at Object.Module._extensions..json (internal/modules/cjs/loader.js:1098:22)
     at Module.load (internal/modules/cjs/loader.js:928:32)
     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
     at Module.require (internal/modules/cjs/loader.js:952:19)
     at require (internal/modules/cjs/helpers.js:88:18)
     at Object.init (/mnt/wiki/server/core/config.js:59:25)
     at Object.<anonymous> (/mnt/wiki/server/index.js:23:16)
     at Module._compile (internal/modules/cjs/loader.js:1063:30) {
   errno: -13,
   syscall: 'open',
   code: 'EACCES',
   path: '/mnt/wiki/package.json'
}
```
What could I do to fix this? 
P.S. Does this have anything to do with adding the admin account to the "User="? I tried entering my admin email but it resulted in a grammar error in the .service file.

**Comment by Power-tile** [comment, marked as answer]
Assigning the root account to the user solved this problem. (i.e. "User=root" in the config file)

**Comment by mckaygerhard** [comment]
puff same case as https://github.com/ralyodio/node-startup/issues/53 and as any windoser solution "runs as administrator" ... real solution https://github.com/ralyodio/node-startup/issues/53#issuecomment-1989926147

usage of root is not unix good practice.. windosers thinks that is like administrator..  and must be not done.. its the only thing 
_shit_ stemd is doing well nowadays .. not allowing root to some tasks
